doc/guides/modules.rst

```diff
@@ -188,11 +129,108 @@ section.
+steering committee and should follow the project licencing and contribution
+guidelines.
+
+A request for a new module should be initiated using an RFC issue in the
+Zephyr project issue tracking system with details about the module and how it
+integrates into the project. If the request is approved, a new repository is
+created and initiated with basic information that would allow submitting code to
+the module project following the project contribution guidelines.
+
+Process
+=======
+
+Follow the following steps to request a new module:
+
+#. Submit an issue with details about the module to be created
+#. propose a name for the repository to be created under the Zephyr project
+   organization on Github.
+#. Maintainers from the Zephyr project will create the repository and initialize
+   it. You will be added as a collaborator in the new repository.
+#. Submit the module content (Code) to the new repository following the
+   guidelines described :ref:`here <modules_using_west>`.
+#. Submit a new entry into the :zephyr_file:`west.yml` with the following
+   information:
+
+   .. code-block:: console
+
+        - name: <name of repository>
+          path: <path to where the repository should be cloned>
+          revision: <ref pointer to module pull request>
+
+
+For example, to add *my_module* to the manifest:
+
+.. code-block:: console
+
+    - name: my_module
+      path: modules/lib/my_module
+      revision: pull/23/head
+
+
+Where 23 in the example above indicated the pull request submitted to the
+*my_module* repository.
+
+.. _changes_to_existing_module:
+
+Changes to existing Modules
+***************************
+
+#. Submit the changes using a pull-request to an existing repository following
+   the :ref:`contribution guidelines <contribute_guidelines>`.
+#. Submit a pull-request changing the entry referencing the module into the
+   :zephyr_file:`west.yml` of the main Zephyr tree with the following
+   information:
+
+   .. code-block:: console
+
+        - name: <name of repository>
+          path: <path to where the repository should be cloned>
+          revision: <ref pointer to module pull request>
+
+
+For example, to add *my_module* to the manifest:
+
+.. code-block:: console
+
+    - name: my_module
+      path: modules/lib/my_module
+      revision: pull/23/head
+
+Where 23 in the example above indicated the pull request submitted to the
+*my_module* repository.
+
+Reviewing and Merging Changes Related to Modules
+*************************************************
+
+Specifying the pull request number in the revision field will allow building the Zephyr main tree with the changes submitted to the
+module. The pull request to master will need to be marked as ``DNM`` (Do Not
+Merge) to allow for the module to first be merged. Once the module is merged,
+the revision will need to be changed either by submitted or by the maintainer
+to the commit hash of the module which reflects the changes.
```

Comment:
> commit hash 

Question: Is the intention that the zephyr manifest will always point to hashes and never to tags of modules?

Reply:
both will work, a tag is a commit after all. we do not create tags for each merged PR in the modules now.

Reply:
I think it would be nice to start tagging the released versions of each module (i.e. tag the version of each module used for Zephyr v1.15.0, etc.) so it would be easy to see using plain `git log` in each module repository what to use. But that's for the next release manager, not regular contributors (@ioannisg, thoughts?)